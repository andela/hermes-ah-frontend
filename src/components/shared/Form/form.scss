@import '../../../assets/stylesheets/index.scss';

.form-wrap {
  height: 86vh;
}

.button-container {
  width: 50%;
  text-align: center;
}

.wrap-cont-login-signup {
  display: flex;
  margin-bottom: 30px;
  height: 55px;
  .signup, .login {
    width: 50%;
    text-align: center;
    cursor: pointer;
    margin: 0;
    border-radius: 0;
    font-weight: 400;
  }

  button.current-login {
    border-radius: 3px 0 0 0;
  }

  button.current-signup {
    border-radius: 0 3px 0 0;
  }

  button.ui.button.login {
    width: 100%;
    background: $primary-color;
    color: #fff;
    font-size: 20px;
  }
  button.ui.button.signup {
    width: 100%;
    background: #fff;
    color: #000;
    font-size: 20px;
  }
  button.ui.button.signup:hover {
    background: #FFF;
    color: $primary-color;
    transition: 0.5s;
  }
}
.form-wrap {
  height: 86vh;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  background: #fcfcfc;
}

.form-cont {
  background: #fff;
  box-shadow: 0 2px 5px #ccc;
  width: 40%;
  border-radius: 3px;
}
form.ui.unstackable.form {
  width: 95%;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  .ui.red.pointing.below.basic.label {
    margin: 0 auto;
  }
  .fields {
    display: flex;
    flex-direction: column !important;
    .ui.input {
      margin: 4px 0;
      height: 50px;
      font-weight: bold;
      font-size: 20px;
    }
  }
  button.ui.button.submit {
    width: 100%;
    background: $primary-color;
    color: #fff;
    height: 55px;
    box-shadow: 0 4px 4px #00000040;
    font-size: 20px;
  }
  .forgot-password {
    color: $primary-color;
    font-weight: 600;
    margin-bottom: 2px;
    text-decoration: none;
    display: flex;
    justify-content: flex-end;
    font-size: 15px;
  }
  input:focus {
    border-color: $primary-color;
  }
}
.hr-bar {
  display: flex;
  align-items: center;
  margin-top: 10px;
  .bar {
    flex: auto;
    border: none;
    height: 1px;
    background-color: #aaa;
  }
  .span-text {
    font-size: 18px;
    padding: 20px 10px;
  }
}
.social-login {
  justify-content: center;
    padding: 12px 0;
    display: flex;
    align-items: center;
    margin-top: -14px;
  i.fab.fa-twitter, i.fab.fa-google, i.fab.fa-facebook-f {
    font-size: 30px;
    background: #55ACEE;
    color: #fff;
    padding: 8px;
    border-radius: 5px;
    cursor: pointer;
  }
  i.fab.fa-google {
    background: #fff !important;
    color: #4384F7 !important;
    font-size: 42px !important;
  }
  i.fab.fa-facebook-f {
    background: #1B4A97 !important;
    padding: 8px 16px !important;
  }
}

::placeholder {
  font-size: 20px;
  color: #857e7e !important;
}
