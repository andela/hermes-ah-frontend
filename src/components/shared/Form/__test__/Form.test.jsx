import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../Form';

describe('HomePage component', () => {
  it('should match snapshot', () => {
    const wrapper = <Form />;
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot renderInput', () => {
    const wrapper = renderer.create(<Form />);
    const inst = wrapper.getInstance();
    expect(inst.renderInput('text', 'text')).toMatchSnapshot();
    expect(inst.renderInput('test1', 'test1', 'password')).toMatchSnapshot();
    expect(inst.renderInput('test2', 'test2', 'email')).toMatchSnapshot();
  });

  it('should match snapshot renderSocialLogin', () => {
    const wrapper = renderer.create(<Form />);
    const inst = wrapper.getInstance();
    expect(inst.renderSocialLogin()).toMatchSnapshot();
  });

  it('should match snapshot renderRuler', () => {
    const wrapper = renderer.create(<Form />);
    const inst = wrapper.getInstance();
    expect(inst.renderRuler()).toMatchSnapshot();
  });

  it('always renders a div', () => {
    const wrapper = shallow(<Form />);
    expect(wrapper.find('div'));
  });
});
