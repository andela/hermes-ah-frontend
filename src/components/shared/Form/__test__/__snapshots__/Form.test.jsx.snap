// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should match snapshot 1`] = `<FormComponent />`;

exports[`HomePage component should match snapshot renderInput 1`] = `
<FormGroup>
  <FormInput
    as={[Function]}
    control={[Function]}
    id="text"
    onChange={[Function]}
    placeholder="text"
    required={true}
    type="text"
  />
</FormGroup>
`;

exports[`HomePage component should match snapshot renderInput 2`] = `
<FormGroup>
  <FormInput
    as={[Function]}
    control={[Function]}
    id="test1"
    onChange={[Function]}
    placeholder="test1"
    required={true}
    type="password"
  />
</FormGroup>
`;

exports[`HomePage component should match snapshot renderInput 3`] = `
<FormGroup>
  <FormInput
    as={[Function]}
    control={[Function]}
    id="test2"
    onChange={[Function]}
    placeholder="test2"
    required={true}
    type="email"
  />
</FormGroup>
`;

exports[`HomePage component should match snapshot renderRuler 1`] = `
<div
  className="hr-bar"
>
  <hr
    className="bar"
  />
  <span
    className="span-text"
  >
    or login with
  </span>
  <hr
    className="bar"
  />
</div>
`;

exports[`HomePage component should match snapshot renderSocialLogin 1`] = `
<div
  className="social-login"
>
  <a
    href="htttwitter.com"
  >
    <i
      className="fab fa-twitter"
    />
  </a>
  <a
    href="google.com.com"
  >
    <i
      className="fab fa-google"
    />
  </a>
  <a
    href="facebook.com"
  >
    <i
      className="fab fa-facebook-f"
    />
  </a>
</div>
`;
